### PR TITLE
Add configure support

### PR DESCRIPTION
**Pull request by sdewitt-newrelic**
## 0.10.0 (2021-06-14)

* Split apart `main.yml` tasks file into multiple files.
* Added tags and support for running tasks individually.
* Updated `default` molecule scenario and added `configure` and `targz`
  scenarios.
* Added additional variables for customizing tarball install.
* Added `nrinfragent_config_file` variable for customizing agent configuration
  path.

### CODE REVIEW

File: requirements.yml [hunk 1]
```diff
@@ -1 +1,2 @@
 ---
```
Comment (JakubKotkowiak) [inline requirements.yml:1]: Some information about this package being required would be nice in the readme (Unless it is there and I simply missed it)

File: tasks/configure_agent.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    - nrinfragent_state != "absent"
+    - nrinfragent_os_name != 'windows'
+    - ansible_python_version is version('3.0', '>=')
+
+- name: setup agent config *NIX
+  yedit:
+    src: "{{ nrinfragent_config_path }}"
+    key: "{{ item.key }}"
+    value: "{{ item.value }}"
+  loop: "{{ lookup('ansible.utils.to_paths', nrinfragent_config) | dict2items }}"
```
Comment (JakubKotkowiak) [inline tasks/configure_agent.yml:25]: Info regarding the requirement of ansible.utils would be useful. It crashed on this step when I was running it locally.

File: requirements.yml [hunk 3]
```diff
@@ -1 +1,2 @@
...
+- src: kwoodson.yedit
```
Comment (JakubKotkowiak) [inline requirements.yml:2]: Please add ansible.utils here as well

**Review by JakubKotkowiak — COMMENTED** [review]
Aside from the comments above it looks ok

File: requirements.yml (hunk 3 above)
Comment (JakubKotkowiak) [reply requirements.yml:2]: Some information about kwoodson.yedit collection being required would be nice in the readme (Unless it is there and I simply missed it)

Comment (sdewitt-newrelic) [reply requirements.yml]: Hi @JakubKotkowiak!  I removed the `kwoodson.yedit` module altogether for concerns about licensing and I created an embedded `merge_yaml.py` module instead.  Please have a look.

File: tasks/configure_agent.yml (hunk 2 above)
Comment (sdewitt-newrelic) [reply tasks/configure_agent.yml]: @JakubKotkowiak with the change to use `merge_yaml.py` (see next comment), this is no longer required.

**Comment by sdewitt-newrelic** [comment]
> One more question, did you check if you can run the molecule converge command?

@JakubKotkowiak Yup.  I have run the `converge` and `verify` commands on all scenarios and for each of the platforms I believe.

**Comment by JakubKotkowiak** [comment]
> > One more question, did you check if you can run the molecule converge command?
> 
> @JakubKotkowiak Yup. I have run the `converge` and `verify` commands on all scenarios and for each of the platforms I believe.

@sdewitt-newrelic Looks good to me! Just a question are you planning any further updates for this? Or is this pretty much ready for the final review?

**Comment by sdewitt-newrelic** [comment]
> > > One more question, did you check if you can run the molecule converge command?
> > 
> > 
> > @JakubKotkowiak Yup. I have run the `converge` and `verify` commands on all scenarios and for each of the platforms I believe.
> 
> @sdewitt-newrelic Looks good to me! Just a question are you planning any further updates for this? Or is this pretty much ready for the final review?

No more updates.  This is ready to go.  Thanks @JakubKotkowiak !

**Comment by JakubKotkowiak** [comment]
I'll recheck it for the last time asap and after that we should see it merged soon, thank you!

**Comment by sdewitt-newrelic** [comment]
Hi @JakubKotkowiak  @AnnaParot any word on getting this integrated?  It now looks like it has conflicts.

**Comment by JakubKotkowiak** [comment]
@sdewitt-newrelic Hello, I will take care of the conflicts on monday and then it should be possible to merge it next week.

**Comment by sdewitt-newrelic** [comment]
Thanks!

**Review by brushknight — COMMENTED** [review]
LGTM, left 1 comment

File: README.md [hunk 4]
```diff
@@ -41,75 +125,213 @@ Use the following snippet to install the agent.
...
+`no` to prevent the service from starting automatically on boot. See additional information about
+valid values in the
+[Ansible service module documentation](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/service_module.html#parameter-state).
 
-##### `nrinfragent_service_state` (Optional)
+##### `nrinfragent_config_file` (Optional)
 
-Specifies the state of the `newrelic-infra` service after install. Defaults to `started`, which ensures the service will be running. You can change it to `stopped` to just install it, without starting it right away.
+Specifies a custom path on the to the New Relic Infrastructure Agent
+[configuration file](sudo curl -s -o /etc/yum.repos.d/newrelic-infra.repo https://download.newrelic.com/infrastructure_agent/linux/yum/el/7/x86_64/newrelic-infra.repo) on the target
```
[the remark this answers is not in the document]
Comment (JakubKotkowiak) [reply README.md:158]: @sdewitt-newrelic it'd seem that this link me be in the incorrect place.

[the remark this answers is not in the document]
Comment (sdewitt-newrelic) [reply README.md]: @JakubKotkowiak @brushknight fixed the link.

**Comment by sdewitt-newrelic** [comment]
I also added a one line update to `merge_yaml.py` to fix a bug I found when merging lists.

**Review by brushknight — APPROVED** [review]
LGTM

**Comment by sdewitt-newrelic** [comment]
Yay!  Thanks @brushknight and @JakubKotkowiak!

**Pull request merged**